src/main/java/edu/harvard/iq/dataverse/PermissionServiceBean.java

```diff
@@ -448,8 +451,9 @@ private boolean isPublicallyDownloadable(DvObject dvo) {
 
             if (!df.isRestricted()) {
                 if (df.getOwner().getReleasedVersion() != null) {
-                    if (df.getOwner().getReleasedVersion().getFileMetadatas() != null) {
-                        for (FileMetadata fm : df.getOwner().getReleasedVersion().getFileMetadatas()) {
+                    List<FileMetadata> fileMetadatas = df.getOwner().getReleasedVersion().getFileMetadatas();
+                    if (fileMetadatas != null) {
+                        for (FileMetadata fm : fileMetadatas) {
```

Comment:
I thought about going a step further and replacing the entire code block with a custom query (it would be faster), but decided to keep it as it is for now, as I didn't want to put any current behavior at risk (I would first isolate and cover the logic with specific tests to do it without risks).

Reply:
In my opinion, it is interesting to consider any effort to replace any existing call to _getFileMetadatas()_.

Reply:
Yes good point. Future work. 😄